### PR TITLE
Rewrite docs

### LINKED ISSUE #211: Better documentation
Nice to have:

- [ ] More examples
- [ ] Stages/Scenes(ex telegraf-flow) advanced usage
- [ ] Extending Telegraf aka implementing middleware
- [ ] Composer magic explained
- [ ] Webhooks. Make requests in response to update stuff
- [ ] API docs for
  - [ ] Markup
  - [ ] Extra
  - [ ] Composer
  - [ ] Router
  - [ ] Stage
  - [ ] Session

### LINKED ISSUE #477: Improved documentation
I really like this project! 
But it is very hard to find the right documentation. 

For instance; 
- What does Markup.forceReply do?
- How can I jump to a specific step in a wizard (…is this even possible)?
- When I look at https://telegraf.js.org/#/?id=extra it points me to the examples. There I see code like this:
`reply('One time keyboard', Markup
    .keyboard(['/simple', '/inline', '/pyramid'])
    .oneTime()
    .resize()
    .extra()
  )`
What does .extra() do? oneTime()? I don't get any clue, anywhere.

Providing more guide-like documentation would help me out a lot. Or maybe links to the Telegram API when appropriate.
As an alternative, I don't mind looking at the source code or examples. But it offers too little information (inline comments) on what methods do and when to use them.

I would be happy to help here, since this looks like a massive job. But I simply don't understand what is going on in many parts of the code.

### LINKED ISSUE #1308: DOcumentation
Put up a decent documentation. The one present now sucks.
I can't even find a proper way to reply to a message (not just sending the text, replying to a message by quoting user).

### PR DESCRIPTION
**Pull request by wojpawlik**
# Description

I'd appreciate reviews from everyone. For your convenience, I built the docs and will update them to reflect my changes: https://telegraf-docs.wojpawlik.vercel.app.

- Fixes #211,
- Fixes #477,
- Closes #1308,
- Advances #1205.

# How Has This Been Tested?

`npm run build:docs`

# Checklist:

- [ ] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,71 +100,19 @@ $ pnpm add telegraf
...
+`ctx` you can see in every example is a [`Context`] instance.
+[`Telegraf`] creates one for each incoming update and passes it to your middleware.
 It contains the `update`, `botInfo`, and `telegram` for making arbitrary Bot API requests,
 as well as shorthand methods and getters.
 
 This is probably the class you'll be using the most.
 
-### Middleware
-
-Middleware is an essential part of any modern framework.
```
Comment (RealPeha) [inline README.md:105]: @wojpawlik Why did you delete this good explanation?

Comment (wojpawlik) [reply README.md:105]: Because I wrote a [better one (IMO) below](https://github.com/telegraf/telegraf/blob/readme/README.md#middleware).

Comment (RealPeha) [reply README.md:105]: And why is it better? As a beginner i didn't understand anything in your explanation

Comment (wojpawlik) [reply README.md:105]: We have a poll which explanation is better: https://t.me/TelegrafJSChat/18314

Personally, I found the old section very roundabout. It has whooping 9 paragraphs, some of them have issues:

> It allows you to modify requests and responses as they pass between the Telegram and your bot.

Very sloppy. It doesn't mention that it's possible to handle or swallow updates as well. Modifying responses is messy until #1267.

> You can imagine middleware as a chain of logic connection your bot to the Telegram request.

I don't understand this at all.

> **`ctx`** is the context for one [Telegram update](https://core.telegram.org/bots/api#update).

A separate section for `Context` says this already.

> **`next`** is a function that is invoked to execute the downstream middleware.
> It returns a `Promise` with a function `then` for running code after completion.

Correct, but should mention `await`.

So, what exactly is missing from the new explanation?

Comment (noneedinmagic) [reply README.md:105]: image
IDK, it feels a bit missing smth, like lack of explanation what is it for, just and explanation of how to use next callback

image
In the old version there is a good explanation about a chain of logic which IMO should be left here for some people who are less experienced with modern JS frameworks and have almost no experience with middleware

Like newcomers to Node 
or
Frontenders who use mostly smth like Vue
or
Backenders who have moderate experience with smth like NestJS and didn't come across writing own middlewares for it

Not all people use vanilla express, you know

And yes, the part about modifying req and res is essential too

Comment (noneedinmagic) [reply README.md:105]: > > It allows you to modify requests and responses as they pass between the Telegram and your bot.
> 
> Very sloppy. It doesn't mention that it's possible to handle or swallow updates as well. Modifying responses is messy until #1267.

1. Then maybe it should be mentioned? About updates. Why delete partial explanation if you can improve it by adding what is missing?

2. If it is messy in any way then maybe it should be mentioned? Like that there is a possibility of modifying the response, but be cautious if you do such thing and such thing and such thing

Comment (noneedinmagic) [reply README.md:105]: >> You can imagine middleware as a chain of logic connection your bot to the Telegram request.

> I don't understand this at all.

Seems like there was some poor translation to English which might need rewriting, not deletion.
Need to think how to rewrite.

File: README.md [hunk 2]
```diff
@@ -101,11 +101,9 @@ or
 $ pnpm add telegraf
 ```
 
-### Bot
+### `Telegraf` class
 
-A Telegraf bot is an object containing an array of middlewares which are composed 
-and executed in a stack-like manner upon request. Is similar to many other middleware systems 
-that you may have encountered such as Express, Koa, Ruby's Rack, Connect.
```
Comment (RealPeha) [inline README.md:101]: @wojpawlik Maybe should be left similarity with others middleware systems? This would help better understand middleware for those who wrote in expressjs etc

**Comment by wojpawlik** [comment]
If there are no more comments, I'll merge on Monday and release 4.0.2.
